### DISCUSSION TITLE
I have a problem about livestreaming connection starlette.websockets.WebSocketDisconnect: 1006

### DISCUSSION THREAD

**Discussion by EunseokEthanYang** [encode/starlette#2383] [Q&A] [answered]
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "D:\backend\app.py", line 22, in websocket_endpoint
    data = await websocket.receive_bytes()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\eunse\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\starlette\websockets.py", line 122, in receive_bytes
    self._raise_on_disconnect(message)
  File "C:\Users\eunse\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\starlette\websockets.py", line 105, in _raise_on_disconnect
    raise WebSocketDisconnect(message["code"])
starlette.websockets.WebSocketDisconnect: 1006

I'm using Fastapi socket connection in Server and using python websocket module in client computer. I was making livestreaming service, but after 30 second I started client camera python program, the error what I said upper happen. the error message upper is printed in Server console. I check by connected with ssh protocol. I use window 11 server by open 8000 port.
Please help me

> [!IMPORTANT]
> - We're using [Polar.sh](https://polar.sh/encode) so you can upvote and help fund this issue.
> - We receive the funding once the issue is completed & confirmed by you.
> - Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/encode/starlette/issues/2382">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/encode/starlette/issues/2382/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/encode/starlette/issues/2382/pledge.svg">
</picture>
</a>

**Comment by coolbeevip** [comment]
same

**Reply by coolbeevip** [reply, marked as answer]
This issue is very similar to mine. 

I am starting a FastAPI web socket service on the server using uvicorn, and the client uses websocket-client (note that this is a synchronous client).

By default, uvicorn's ping/pong interval is 20 seconds. When the websocket-client blocks for more than 20 seconds during sending, it triggers a failure in uvicorn's ping check and actively disconnects with error code 1006.

The simple solution is to set the ws_ping_interval interval.

```
uvicorn.run(app, ws_ping_interval=600, ws_ping_timeout=60)
```

**Reply by ugyballoons** [reply]
Why does setting the interval to 10 minutes and the timeout to 1 minute work?
I can see in logs that my server is generating, that client disconnection is now handled cleanly.
I'm grateful that there's a simple solution, I just don't understand it.

**Reply by coolbeevip** [reply]
Uvicorn employs the WebSocket ping mechanism at a default interval of 20 seconds to ascertain the health of the connection.

If you are utilizing a synchronous WebSocket client (such as websocket-client) to transmit data persistently over a period exceeding 20 seconds, this operation may be interrupted. The interruption is due to Uvicorn's default behavior, where it interprets the absence of ping responses during prolonged data transmission as a connectivity issue, consequently leading to the termination of the client connection.

In my specific application context, the ping interval has been configured to 10 minutes. This adjustment is made based on the requirement that continuous data transmission is not expected to exceed this duration within the scope of my business operations.

For details on adjusting the websocket ping interval (ws-ping-interval), refer to Uvicorn's documentation at https://www.uvicorn.org/settings/.

**Reply by ugyballoons** [reply]
Of course! & thank you for your reply.

I skipped over your important note that mentioned it's a synchronous client that's running which is fundamental to your solution.
I've applied the setting of the interval and timeout to an **asynchronous** client running in FastAPI via uvicorn and found that unexpected disconnections, that were raising exceptions outside of the handling blocks in the endpoint, are now being handled cleanly.

It's something that appears to be working purely by coincidence & that I wouldn't have dreamt of trying had I read your post properly!

**Reply by EunseokEthanYang** [reply]
Thank you for replying me 😃😃😃

**Reply by coolbeevip** [reply]
Good luck to everyone

**Reply by weng8232** [reply]
solved!  love you~~,you are my god!!!
